src/jlgen.jl

```diff
@@ -480,8 +480,32 @@ end # HAS_INTEGRATED_CACHE
 ## codegen/inference integration
 
 function ci_cache_populate(interp, cache, mi, min_world, max_world)
-    if VERSION >= v"1.12.0-DEV.15"
-        inferred_ci = CC.typeinf_ext_toplevel(interp, mi, CC.SOURCE_MODE_FORCE_SOURCE) # or SOURCE_MODE_FORCE_SOURCE_UNCACHED?
+    @static if VERSION >= v"1.12.0-DEV.1434"
+        # see typeinfer.jl: typeinf_ext_toplevel
+        ci = CC.typeinf_ext(interp, mi, CC.SOURCE_MODE_NOT_REQUIRED)
+        inspected = IdSet{CodeInstance}()
+        tocompile = CodeInstance[ci]
```

Comment:
I removed the check for `use_const_api` here since that seems handled below already (this looks like a bug/typo in Base).

Reply:
They handled unrelated things. The first determines whether you want to generate code for it (Base does not) and the second determines what code it would have to use if it was forced to generate code anyways (which well designed callers normally wouldn't do because of the first check and due to inlining implementation quality choices)

Reply:
OK, so I guess in our case (where we need to generate code) it's fine like I did.